pkgs/by-name/en/ente-web/package.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  nodejs,
+  yarnConfigHook,
+  yarnBuildHook,
+  extraBuildEnv ? null,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ente-web";
+  version = "0.9.16";
+
+  src =
+    fetchFromGitHub {
+      owner = "ente-io";
+      repo = "ente";
+      sparseCheckout = [ "web" ];
+      rev = "photos-v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-DqfUUXY79CndEqPT8TR4PasLtaSCtqZaV2kp10Vu4PQ=";
+    }
+    + "/web";
```

Comment:
I'm using `finalAttrs` instead of `rec` here, thinking this was the recommendation where. I'm not sure how to use `name` with that.

Reply:
> I'm using `finalAttrs` instead of `rec` here, thinking this was the recommendation where. I'm not sure how to use `name` with that.

Simple: `finalAttrs.src.name`

Reply:
@doronbehar When using the pattern with sourceRoot as above, it fails to build. I changed it like this:
```
  src = fetchFromGitHub {
    owner = "ente-io";
    repo = "ente";
    sparseCheckout = [ "web" ];
    rev = "photos-v${finalAttrs.version}";
    hash = "sha256-vucWtu7abQYFDKStAAzE2ofG87AZD2vEdNmPMnpQyeg=";
  };

  sourceRoot = "${finalAttrs.src.name}/web";
```
And get the following error:

```

syntax: prefetch-yarn-deps [path to yarn.lock] [options]

Options:
  -h --help         Show this help
  -v --verbose      Verbose output
  --builder         Only perform the download to current directory, then exit
error: builder for '/nix/store/sgahsqi02jirhcmny9xhsqwid2g0n0is-offline.drv' failed with exit code 1;
       last 12 log lines:
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       >
       > syntax: prefetch-yarn-deps [path to yarn.lock] [options]
       >
       > Options:
       >   -h --help         Show this help
       >   -v --verbose      Verbose output
       >   --builder         Only perform the download to current directory, then exit
       For full logs, run 'nix log /nix/store/sgahsqi02jirhcmny9xhsqwid2g0n0is-offline.drv'.
error: 1 dependencies of derivation '/nix/store/vc634i1h4cp0nrzbp1vyyphn20rw8jng-ente-web-0.9.16.drv' failed to build
```

Reply:
See the current implementation.

Reply:
> See the current implementation.

Thanks for clarifying! I fixed the hash and rebuild the package, it builds correctly.